Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,96 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            bool nonExistantCheckWarning = false;
+
+            try
+            {
+                if (externalBoundary.IIsSelfIntersecting())
+                {
+                    Reflection.Compute.RecordError("The provided externalBoundary is selfintersecting.");
+                    return null;
+                }
+            } catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (internalBoundaries.Any(x => x.IIsSelfIntersecting()))
+                {
+                    Reflection.Compute.RecordError("At least one of the internalBoundaries is selfintersecting.");
+                    return null;
+                }
+            }
+            catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (internalBoundaries.Any(x => !externalBoundary.IIsContaining(x)))
+                {
+                    Reflection.Compute.RecordError("At least one of the internalBoundaries is not contained by the externalBoundary.");
+                    return null;
+                }
+            } catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            bool unionDone = false;
+            try
+            { 
+                for (int i = 0; i < internalBoundaries.Count; i++)
+                {
+                    for (int j = i + 1; j < internalBoundaries.Count; j++)
+                    {
+                        if (internalBoundaries[i].ICurveIntersections(internalBoundaries[j]).Count != 0)
+                        {
+                            internalBoundaries = Compute.BooleanUnion(internalBoundaries).Cast<ICurve>().ToList();
+                            Reflection.Compute.RecordWarning("InternalBounderies were overlapping, BooleanUnion has been used to get non overlapping regions.");
+                            i = internalBoundaries.Count;
+                            j = i;
+                            unionDone = true;
+                        }
+                    }
+                }
+            }
+            catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (!unionDone)
+                {
+                    for (int i = 0; i < internalBoundaries.Count; i++)
+                    {
+                        for (int j = 0; j < internalBoundaries.Count; j++)
+                        {
+                            if (i != j && internalBoundaries[i].IIsContaining(internalBoundaries[j]))
```

Comment:
Uhm, I didn't get this. Why aren't we doing adding this check in the previous attempt to boolean unite the internal buondaries?

Reply:
The previous check does not catch the case of a internal boundary encapsulating without intersecting another. 

Keeps this from being the curve definition:
![image](https://user-images.githubusercontent.com/53530468/81913912-84aab800-95c8-11ea-8e0b-5a25a6809360.png)
which would at least mess with the area method

Reply:
But can't the check happen in the for loops above?

Reply:
The for loop above is not the same as it only checks each pair once while this one checks every pair in both directions